### DISCUSSION TITLE
4x 3fach-Stufenschalter funktionieren nicht

### DISCUSSION THREAD

**Discussion by roscje** [camueller/SmartApplianceEnabler#621] [Q&A]
Hallo Forum, Hallo camueller,
ich muss sagen, SAE ist einfach nur eine tolles Projekt, das SMA-SmartHome Universum zu erweitern. Vielen Dank an camueller.

Seit 2019 betreibe ich SAE auf einem Pi0 erfolgreich.

Eine zweite Anlage soll nun Heizstäbe für Pufferspeicher intelligent steuern.
Die PVA ist etwas größer (50kWp), so dass sich das Ganze auch lohnt. Eine Batterie ist auch in der Anlage vorhanden.
Es sollen vier Speicher geladen werden, die mit jeweils einem Heizstab ausgestattet sind. Dabei soll die Priorität eingehalten werden.

Prio1: Speicher WW   Stufe1: 2kW, Stufe 2:4kW, Stufe3;6kW
Prio2: Speicher 1HZ  Stufe1: 1,5kW, Stufe 2:3kW, Stufe3;4,5kW
Prio3: Speicher 2HZ  Stufe1: 1,5kW, Stufe 2:3kW, Stufe3;4,5kW
Prio4: Speicher 3HZ  Stufe1: 1,5kW, Stufe 2:3kW, Stufe3;4,5kW

Es liegt nahe dafür vier Stufenschalter mit jeweils 3 Stufen im SAE anzulegen. Die Prio kann über das Sunnyportal für jeden Verbraucher eingestellt werden. Geschaltet werden die Heizstäbe mit den GPIOs eines RPi4B über eine Relaisplatine. Der Verbrauch wird nicht erfasst, da bei Heizstäben relativ stabil zur Nennleistung. Es soll auschließlich mit Überschuss geheizt werden.

Es ist alles installiert und in Betrieb genommen. Der HM verhält sich aber nicht wie angenommen. 
Die Heizstäbe werden zugeschalten, obwohl die Batterie noch nicht vollgeladen ist. Prognosebasiertes Laden ist deaktiviert, priorisiertes Laden der Batterie ist im SunnyPortal aktiviert. Es wird sogar die Batterie entladen, um die Speicher zu bedienen. HM wurde bereits neugestartet.

Ich wollte es dem HM leicht machen und habe keine minimalen und maximalen Ein-und Ausschaltzeiten vergeben, aber bin mir nicht sicher, ob es beim Stufenschalter doch notwendig ist. Es soll schwellwertbasiert geschalten werden, also ohne Planung des HM.

Da das Verhalten überhaupt noch nicht nachvollziehbar ist, ist es vielleicht besser zu klären, ob die Konfig stimmt. Danach kann das Log untersucht werden.
Eventuell gibt es auch einen anderen Ansatz für meine Problemstellung
Irgendwie komme ich nicht weiter.

Grüße 
roscje

**Comment by camueller** [comment]
Ohne Vebrauchserfassung wird das nicht funktionieren aktuell, insbesondere bei Verbrauchern mit variabler Leistungsaufnahme.

**Comment by roscje** [comment]
Ok. Das ist mir neu.
Ich dachte wirklich, dass SAE den Wert der Stufe des Verbrauchers als aktuellen Wert dem HM meldet, wenn kein Zähler angegeben wird. So ist es doch auch bei Verbrauchern mit "normalen" Schalter, oder?
Die Verbraucherbilanz im Portal zeigt die maximale Leistungsaufnahme komplett über das ganze Zeitfenster. Wird dann, wenn kein aktueller Wert gesendet wird, immer die maximale Leistung als Bezug genommen?
Kann es dann funktionen, wenn ich S0-Zähler noch nachrüste? Auch in Bezug auf die Priorität im HM?

Vielen Dank

**Comment by camueller** [comment]
> Die Verbraucherbilanz im Portal zeigt die maximale Leistungsaufnahme komplett über das ganze Zeitfenster.

Genau. Und das ist in Deinem Fall ja nicht sinnvoll. Klar könnte man das programmieren, aber das wäre ein Sonderfall und mindestens für PWM-Schalter würde es nicht funktionieren ohne Zähler.

>Wird dann, wenn kein aktueller Wert gesendet wird, immer die maximale Leistung als Bezug genommen?

Ja, weil das für alle Geräte funktioniert, nicht nur für solche mit variabler Leistungsaufnahme.

> Kann es dann funktionen, wenn ich S0-Zähler noch nachrüste? Auch in Bezug auf die Priorität im HM?

Der SHM muss immer wissen, wie hoch die aktuelle Leistungsaufnahme ist, damit er sinnvoll planen kann. Dafür kannst Du beliebige Zähler verwenden, solange sie vom SAE unterstützen werden und ohne zusätzliche Rechenoperationen den Zählerstand oder die Leistungsaufnahme liefern können.

**Comment by camueller** [comment]
Nachtrag: 4 Stufenschalter bedeutet: 4 Geräte. Also brauchst Du auch 4 Zähler.

**Comment by roscje** [comment]
Danke für deine schnelle Antwort.
Genau 4 Zähler ist ein größerer Kostenfaktor und die Verteilung gibt nicht mehr 16TE Platz her.
Ich werde den Stufenschalter nicht verwenden, wenn es nur mit Zähler sinnvoll funktioniert.

12 einzelne Schaltverbraucher sollten die gewünschte Funktion auch erfüllen erfüllen.

Schade, ich  wollte wirklich mal die Stufenschalter und später auch PWM produktiv nutzen.

Nochmals vielen Dank für deine Hilfe.

**Comment by camueller** [comment]
Die Heizstäbe sind ja bestimmt nicht dort, wo Dein Verteilerschrank ist. Ich habe bei mir im Heizungskeller einen kleinen Unterverteiler mit Hutschiene. So etwas bekommt man ab 20 Euro und wenn man es selber verdrahtet kommt nix weiter dazu.

**Reply by roscje** [reply]
Doch, die Leitungen für die Heizstäbe gehen alle aus einer Verteilung weg. So sind nun 12 Relais am SAE mit 4 unterschiedlichen Prioritäten in Betrieb. 
Meinst du einen Zähler für 20 Euro? Der wäre dann einphasig oder? Ich bräuchte dann 12 Stück!
Im Moment funktioniert es auch ohne.

Vielleicht ist es sinnvoll, in der Doku zu ergänzen, dass der Stufenschalter auch einen Zähler benötigt.
z.B. so:
Auch wenn die Leistung pro Stufe fest vorgegeben wird, benötigt der Homemanager die momentane Leistungsaufnahme des Stufenschalters. Es ist ein Zähler für den Verbraucher erforderlich!

**Comment by camueller** [comment]
> Meinst du einen Zähler für 20 Euro?

Nein, einen Unterverteiler mit einer Schiene. Zähler bräuchtest Du wohl 3-phasige - die kosten knapp 50 Euro das Stück.
